activiti-cloud-query-service/activiti-cloud-starter-query/src/test/java/org/activiti/cloud/starter/tests/QueryProcessInstancesEntityIT.java

```diff
@@ -626,6 +626,102 @@ public void shouldGetProcessInstanceInitiatedByCurrentUser() {
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.anAssignedTask("ATask", "arandomuser", thirdProcessInstanceWithRandomUserAssignedToTask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserAssignedToATask.getId(),
+            secondProcessInstanceWithTestUserAssignedToATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsTaskCandidate() {
+        ProcessInstance firstProcessInstanceWithTestUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "testuser", firstProcessInstanceWithTestUserCandidateOfATask);
+
+        ProcessInstance secondProcessInstanceWithTestUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "testuser", secondProcessInstanceWithTestUserCandidateOfATask);
+
+        ProcessInstance thirdProcessInstanceWithRandomUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "arandomuser", thirdProcessInstanceWithRandomUserCandidateOfATask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserCandidateOfATask.getId(),
+            secondProcessInstanceWithTestUserCandidateOfATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsTaskAssigneeOrCandidate() {
+        ProcessInstance firstProcessInstanceWithTestUserAssignedToATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.anAssignedTask("ATask", "testuser", firstProcessInstanceWithTestUserAssignedToATask);
+
+        ProcessInstance secondProcessInstanceWithTestUserCandidateOfATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.aTaskWithUserCandidate("ATask", "testuser", secondProcessInstanceWithTestUserCandidateOfATask);
+
+        ProcessInstance thirdProcessInstanceWithRandomUserAssignedToATask = processInstanceBuilder
+            .aRunningProcessInstanceWithInitiator("process", "arandomuser");
+
+        taskEventBuilder.anAssignedTask("ATask", "arandomuser", thirdProcessInstanceWithRandomUserAssignedToATask);
+
+        eventsAggregator.sendAll();
+
+        shouldGetProcessInstancesList(List.of(firstProcessInstanceWithTestUserAssignedToATask.getId(),
+            secondProcessInstanceWithTestUserCandidateOfATask.getId()));
+    }
+
+    @Test
+    public void shouldGetProcessInstancesWhenCurrentUserIsInitiatorOrTaskAssigneeOrCandidate() {
```

Comment:
Same as before. Should we just cover the case when the user is Initiator here?

Reply:
There is a test when the user is initiator. I wanted to make sure that query works using or.

Reply:
I had exactly the same thought as @Giovanni007 .  I'm wondering if having only this test that is covering the three cases is not enough. But if you prefer to keep them all to make sure all the scenarios are working it's also fine.